src/test/groovy/com/avioconsulting/mule/linter/rule/configuration/ConfigFileNamingRuleTest.groovy

```diff
@@ -0,0 +1,93 @@
+package com.avioconsulting.mule.linter.rule.configuration
+
+import com.avioconsulting.mule.linter.TestApplication
+import com.avioconsulting.mule.linter.model.Application
+import com.avioconsulting.mule.linter.model.CaseNaming
+import com.avioconsulting.mule.linter.model.Rule
+import com.avioconsulting.mule.linter.model.RuleViolation
+import spock.lang.Specification
+
+class ConfigFileNamingRuleTest extends Specification {
+
+    private final TestApplication testApp = new TestApplication()
+    private Application app
+
+    def setup() {
+        testApp.create()
+        testApp.addPom()
+        testApp.addConfig()
+        testApp.buildConfigContent('camelCase.xml', '')
+        testApp.buildConfigContent('PascalCase.xml', '')
+        testApp.buildConfigContent('snake_case.xml', '')
+        testApp.buildConfigContent('kebab-case.xml', '')
+    }
+
+    def cleanup() {
+        testApp.remove()
+    }
+
+    @SuppressWarnings(['MethodName', 'MethodReturnTypeRequired'])
+    def 'default config file naming convention check'() {
+        given:
+        Rule rule = new ConfigFileNamingRule()
+
+        when:
+        Application app = new Application(testApp.appDir)
+        List<RuleViolation> violations = rule.execute(app)
+
+        then:
+        !violations.any { it.fileName.contains('kebab-case.xml')}
```

Comment:
Each test should also validate either the number of violations expected (3) or the file names that were found with violations.  We know that kebab-case.xml wasn't a violation, but what if none were?